pkg/controller/apiextensions/claim/api.go

```diff
@@ -105,25 +57,32 @@ func NewAPIBinder(c client.Client, t runtime.ObjectTyper) *APIBinder {
 
 // Bind the supplied claim to the supplied composite.
 func (a *APIBinder) Bind(ctx context.Context, cm resource.CompositeClaim, cp resource.Composite) error {
-	existing := cp.GetClaimReference()
-	proposed := meta.ReferenceTo(cm, resource.MustGetKind(cm, a.typer))
-
-	if existing != nil && (existing.Namespace != proposed.Namespace || existing.Name != proposed.Name) {
-		return errors.New(errBindConflict)
+	existing := cm.GetResourceReference()
+	proposed := meta.ReferenceTo(cp, resource.MustGetKind(cp, a.typer))
+	if existing != nil && !cmp.Equal(existing, proposed, cmpopts.IgnoreFields(corev1.ObjectReference{}, "UID")) {
```

Comment:
This would be a nice helper to have in `crossplane-runtime` as a helper at some point :)

Reply:
Possibly, though ideally we'll be rid of ObjectReference soon per https://github.com/crossplane/crossplane-runtime/issues/49. I don't think our own equivalent types include a UID.